x/evm/types/utils.go

```diff
@@ -52,23 +58,39 @@ func rlpHash(x interface{}) (hash ethcmn.Hash) {
 	return hash
 }
 
+// ResultData represents the data returned in an comsos-sdk/types.Result
+type ResultData struct {
+	Address ethcmn.Address
+	Bloom   ethtypes.Bloom
+	Logs    []*ethtypes.Log
+	Ret     []byte
+}
+
 // EncodeReturnData takes all of the necessary data from the EVM execution
 // and returns the data as a byte slice
-func EncodeReturnData(addr ethcmn.Address, bloom ethtypes.Bloom, evmRet []byte) []byte {
-	// Append address, bloom, evm return bytes in that order
-	returnData := append(addr.Bytes(), bloom.Bytes()...)
-	return append(returnData, evmRet...)
+func EncodeResultData(data *ResultData) ([]byte, error) {
+	return rlp.EncodeToBytes(data)
```

Comment:
any reason to use RLP instead of amino in all these functions?

Reply:
it just seemed straightforward to use, I could switch it to amino if that's preferred

Reply:
let's use amino everywhere. I'll later do a migration to protobuf from amino

Reply:
do you mean specifically for this or the whole app? If you are migrating to something it may make sense to switch the protocol to rlp to have the tx and block hashes match Ethereum? (Or am I misunderstanding what you meant by this?)

Reply:
amino is being deprecated in favor of protocol buffers. I'm not that familiar with RLP. My main interest in having protobuf is:

- From the next version, it will be the standard of the SDK. This means that existing modules on the EthermintApp will all be protobuf encoded (for both state and client). Thus, having only the EVM module with RLP encoding would be odd imho. I'd like to have some performance discussions/benchmarking about this or a design document
- It will be supported by IBC (proto schemas)

Reply:
hmm didn't realize IBC was protobuf specific. Definitely a trade off, but the reason I mention is that the discrepancy in hashes/ underlying encoding protocol is why Ethers.js doesn't work with our node. It was a vague plan to update at some point to RLP from amino but was not seriously explored before. 

You guys will definitely have to weigh the pros and cons but just figured I'd give my 2 cents of why it would be beneficial for RLP in terms of compatibility with certain dev tooling and expectations transitioning from Ethereum.

Reply:
okay I just updated this to amino